templates/problem/voting-controls.html

```diff
@@ -0,0 +1,65 @@
+<hr style="padding-top: 0.1em">
+
+<div class="vote-form" id="id_vote_form_box">
+    <div id="id_has_voted_prompt">
+        <span class="vote-form-info">{{ _('Current Vote:') }}</span>
+        <span id="id_current_vote_value" class="vote-form-value"></span>
+        <br/>
+        <br/>
+        <span class="vote-form-text">{{ _('Change your vote below!') }}</span>
+        <br/>
+        <br/>
+    </div>
+    <div id="id_has_not_voted_prompt">
+        <span class="vote-form-text">{{ _("Cast your vote on this problem's weighting below!") }}</span>
+        <br/>
+        <br/>
+    </div>
+    <form id="id_delete_vote_form">
+        {% csrf_token %}
+    </form>
+    <form id="id_vote_form">
+        {% csrf_token %}
+        <span class="vote-form-info">{{ _('Points') }}</span>
+        <input id="id_points" class="vote-form-value" type="number" step="1"
+               min="{{ min_possible_vote }}" max="{{ max_possible_vote }}" name="points">
+        <br/>
+        <div id="id_points_error_box">
+            <br/>
+            <span id="id_points_error" class="voting-form-error"></span>
+            <br/>
+        </div>
+        <br/>
+        <span class="vote-form-info">{{ _('Note') }}</span>
+        <br/>
+        <textarea id="id_note" class="vote-form-text" name="note" rows="6" cols="80"></textarea>
+        <br/>
+        <div id="id_note_error_box">
+            <br/>
+            <span id="id_note_error" class="voting-form-error"></span>
+            <br/>
+        </div>
+        <br/>
+        <div>
+            <input type="submit" id="id_vote_form_submit_button" class="vote-button" value="{{ _('Vote') }}">
+            <input type="submit" id="id_delete_vote_form_submit_button" value="{{ _('Delete vote') }}" form="id_delete_vote_form">
+        </div>
+    </form>
+</div>
+
+<div>
+    <a class="form-button" id="id_vote_button"></a>
+</div>
+
+{% if vote is not none %}
+    {{ vote.note|json_script('voted_note') }}
+    <script>
+        set_default_vote_data({{ vote.points }});
+    </script>
+{% endif %}
+
+<script>
+    set_translated_strings('{{ _('Change vote') }}', '{{ _('Vote on problem points') }}', "{{ _("A short justification for your vote to this problem's points value.") }}");
+    set_voting_urls('{{ url('vote', object.code) }}', '{{ url('delete_vote', object.code) }}');
+    startVotingControls();
+</script>
```

Comment:
Hm... I want someone elses opinion: does having this at the bottom of the file affect things? I.e., will we display a non-translated string for longer than if this were at the top?

Reply:
I think the translation is evaluated server-side, so it should not affect anything.

Reply:
Not what I mean. Since this script is low on the page, the JS functions don't get called for a while.

Reply:
As it stands, it shouldn't display anything until the JS runs.

Reply:
Alright.